protocols/kad/src/behaviour.rs

```diff
@@ -802,6 +909,14 @@ pub enum KademliaOut {
         /// List of peers ordered from closest to furthest away.
         closer_peers: Vec<PeerId>,
     },
+
+    /// Result of a `GET_VALUE` query
+    GetValueResult {
+        /// The result that we have probably received
+        result: Option<(Multihash, Vec<u8>)>,
+        /// List of peers ordered from closes to furthest from the key
+        closer_peers: Vec<PeerId>,
```

Comment:
I think a `GetValueResult` should actually not contain `closer_peers` if a value is found, i.e. it should either contain the value, or the closest peers. This allows to short-circuit `get_value` if the key is found in the local storage (and `Q=1`).